operator/internal/custom/istio_client.go

```diff
@@ -15,46 +17,84 @@ import (
 )
 
 const (
-	firstElementIdx = 0
-	contractVersion = "1"
+	firstElementIdx    = 0
+	contractVersion    = "1"
+	virtualServiceName = "kcp-events"
 )
 
 type IstioClient struct {
 	istioclient.Interface
+	// IstioGateway string
+	IstioGateway client.ObjectKey
 }
 
-func NewVersionedIstioClient(cfg *rest.Config) (*IstioClient, error) {
+func NewVersionedIstioClient(cfg *rest.Config, istioGw string) (*IstioClient, error) {
+	gwStrings := strings.Split(istioGw, "/")
+	if len(gwStrings) != 2 {
+		return nil, fmt.Errorf("error validating gateway config: expected (namespace/name), got: (%s)", istioGw)
+	}
 	cs, err := istioclient.NewForConfig(cfg)
 	if err != nil {
 		return nil, err
 	}
 	return &IstioClient{
 		Interface: cs,
+		IstioGateway: client.ObjectKey{
+			Namespace: gwStrings[0],
+			Name:      gwStrings[1],
+		},
 	}, nil
 }
 
-func (c *IstioClient) getVirtualService(ctx context.Context, vsObjectKey client.ObjectKey,
+func (c *IstioClient) getOrCreateVirtualService(ctx context.Context, obj *v1alpha1.Watcher,
 ) (*istioclientapi.VirtualService, error) {
-	virtualService, err := c.NetworkingV1beta1().
-		VirtualServices(vsObjectKey.Namespace).
-		Get(ctx, vsObjectKey.Name, metav1.GetOptions{})
-	if err != nil {
+	var err error
+	var virtualService *istioclientapi.VirtualService
+	virtualService, err = c.NetworkingV1beta1().
+		VirtualServices(c.IstioGateway.Namespace).
+		Get(ctx, virtualServiceName, metav1.GetOptions{})
+	if client.IgnoreNotFound(err) != nil {
 		return nil, fmt.Errorf("failed to fetch virtual service %w", err)
 	}
+	if kerrors.IsNotFound(err) {
```

Comment:
consider rename it to `apierrors`, that's how usually the naming defined in [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/interfaces.go#L22)